OpcUaClient/OpcUaInterface.hpp

```diff
@@ -218,14 +232,193 @@ class OpcUaWrapper : public OpcUaInterface {
+
+					selectClauses[2].typeDefinitionId = UA_NODEID_NUMERIC(0, UA_NS0ID_BASEEVENTTYPE);
+					selectClauses[2].attributeId = UA_ATTRIBUTEID_VALUE;
+    				selectClauses[2].browsePathSize = 1;
+    				selectClauses[2].browsePath = (UA_QualifiedName*)
+       				UA_Array_new(selectClauses[2].browsePathSize, &UA_TYPES[UA_TYPES_QUALIFIEDNAME]);
+    				if(!selectClauses[2].browsePath) {
+        				UA_SimpleAttributeOperand_delete(selectClauses);
+    				}
+
+    				selectClauses[2].browsePath[0] = UA_QUALIFIEDNAME_ALLOC(0, "Changes");    
+
+					filter.selectClauses = selectClauses;
+    				filter.selectClausesSize = 3;
+					// Content Filter
+					UA_ContentFilter contentFilter;
+					UA_ContentFilter_init(&contentFilter);
+
+					UA_ContentFilterElement contentFilterElement;
+					UA_ContentFilterElement_init(&contentFilterElement);
+
+					UA_LiteralOperand literalOperand;
+					UA_LiteralOperand_init(&literalOperand);
+					UA_Variant v;
+					UA_NodeId generalModelChangeEventNodeId = UA_NODEID_NUMERIC(0, UA_NS0ID_GENERALMODELCHANGEEVENTTYPE);
+					UA_Variant_setScalar(&v, &generalModelChangeEventNodeId, &UA_TYPES[UA_TYPES_NODEID]);
+					literalOperand.value = v;
+
+					contentFilterElement.filterOperator = UA_FILTEROPERATOR_OFTYPE;
+					contentFilterElement.filterOperandsSize = 1;
+					contentFilterElement.filterOperands = (UA_ExtensionObject*)
+					UA_Array_new(contentFilterElement.filterOperandsSize, &UA_TYPES[UA_TYPES_EXTENSIONOBJECT]);
+					UA_ExtensionObject extensionObject;
+					UA_ExtensionObject_setValue(&extensionObject, &literalOperand, &UA_TYPES[UA_TYPES_LITERALOPERAND]);
+					contentFilterElement.filterOperands[0] = extensionObject;
+					if(!contentFilterElement.filterOperands) {
+						UA_ContentFilterElement_delete(&contentFilterElement);
+					}
+
+					contentFilter.elementsSize = 1;
+					contentFilter.elements = (UA_ContentFilterElement* ) UA_Array_new(contentFilter.elementsSize, &UA_TYPES[UA_TYPES_CONTENTFILTERELEMENT]);
+					contentFilter.elements[0] = contentFilterElement;
+					filter.whereClause = contentFilter;
+
+
+    				item.requestedParameters.filter.encoding = UA_EXTENSIONOBJECT_DECODED;
+    				item.requestedParameters.filter.content.decoded.data = &filter;
+    				item.requestedParameters.filter.content.decoded.type = &UA_TYPES[UA_TYPES_EVENTFILTER];
+
+    				UA_MonitoredItemCreateResult result = UA_Client_MonitoredItems_createEvent(client, subId, UA_TIMESTAMPSTORETURN_BOTH, item, this, handler_events, NULL);
+
+					if(result.statusCode == UA_STATUSCODE_GOOD) {
+       					LOG(INFO) << "Created MonitoredItem";
+						return std::make_shared<Dashboard::IDashboardDataClient::EventSubscriptionHandle>(result.monitoredItemId, item.requestedParameters.clientHandle);
+					} else {
+						LOG(ERROR) << "Unable to create MonitoredItem";
+						return NULL;
+					}
+
+    			} else {
+					return NULL;
+				}	
+				return NULL;
```

Comment:
Looks like these indentations can be easily avoided by earlier returns.
In C++ instead of returning NULL, nullptr is the better choice most of the time.

Reply:
Done.